syntax = "proto3";

package mrpc;

enum RpcErrCode {
    RPC_ERR_SUCCESS = 0;
    RPC_ERR_NOTFOUND = 1;
    RPC_ERR_TIMEOUT = 2;
}

// enum RpcServiceName {
//    RPC_ACCOUNT_SERVICE = 0;
// }

message AccountInfo {
    string name = 1;
    string description = 2;
}

message InfoUpdateRequest {
    string service_name = 1;
    string method_name = 2;
    AccountInfo info = 3;
    string new_description = 4;
}

message InfoUpdateResponse {
    AccountInfo info_ret = 1;
    RpcErrCode err_code = 2;
}

// service RpcAccountService {
//     rpc CallAccountService (InfoUpdateRequest) returns (InfoUpdateResponse);
// }